_drafts/monitoring-battery-life.md

```diff
@@ -0,0 +1,500 @@
+
+## Measuring power consumption of the hardware
+
+Let's start trying to build a model for how long our hardware device will last when operating on battery power. The first step to take is to measure how much power the hardware consumes at a base level. This is when the device is operating in all three major states: minimal capacity, normal capacity, and under strenuous load.
+
+Having all three of these power consumption profiles help paint a picture of how much power this piece of hardware may consume, and therefore how quickly it might drain a battery.
+
+### Power Profiles for each component
+
+The first step is to create baselines for how much power each component will consume **over time**. Although the spec sheets from the hardware vendor are great and will generally tell you power consumption - they aren’t always accurate, and different components from different batches will have different power usage characteristics.
+
+The general steps to accomplishing this are:
+
+- Ensure that your development boards have power rails so that you can isolate many of the components and you can easily attach probes to them.
+- Get a nice multimeter that can measure µA and mA.
+- Write a special firmware that instruments a single component in various power modes, and determine the current draw for each mode. This should be the simplest possible firmware and ideally only contain driver code.
+- Rinse and repeat with every component that could have a big impact on power consumption.
+
+For example, the accelerometer was a component in a wearable device that could consume a lot of power if left in the wrong state for periods of time. We wrote a firmware that would set it to different sampling rates and record the current consumption and then used this to determine at what sampling rate we could keep the accelerometer while still achieving our desired 7-day battery life.
+
+### State of Charge (SoC) over Time
+
+Once we determine whether or not our hardware can successfully meet our minimum requirements for battery life, now we need to put this to the test from the other direction. The primary thing we’ll look into now is how to measure the state of charge (SoC) of the battery over time to ensure that the device can last as long as required.
+
+One of the most common and simplest ways to measure the current capacity within a battery, or the SoC of a device, is by measuring the voltage of the battery during a known and consistent power draw, but it's not the only way.
+
+#### Coulomb Counting with Fuel Gauges
+
+At Pebble, we had a fuel gauge on one of the watches. A fuel gauge is a nifty hardware component that can indicate the battery’s SoC and health. It can understand the battery’s current capacity in two ways: measuring the voltage from the battery and learning the patterns over time, and also Coulomb counting, which measures how much current passes in and out of the battery over time.
```

Comment:
Not sure if I'm reading this wrong, but isn't that three ways it measures batter capacity?

Reply:
I'll make it cleaner